### ISSUE TITLE
Issue a warning when unsupported characters are used in wildcard names

### ISSUE THREAD

**Issue by SichongP** [snakemake/snakemake#1579]
**Is your feature request related to a problem? Please describe.**

Wildcard names only allow [a-zA-Z0-9_] and when an unsupported character is included in a wildcard name, it is silently ignored ([`get_wildcard_names`](https://github.com/snakemake/snakemake/blob/fe50b5ca4580a0210343bfcee559831888e80a87/snakemake/io.py#L705-L706) returns an empty set). This can lead to confusing error messages like in issue #1349

**Describe the solution you'd like**
Issue a warning message, something along the lines of "there appears to be an unsupportedcharacter in what seems like a wildcard name. Wildcard names can only include alphanumeric characters and underscore"

**Describe alternatives you've considered**
Alternative is silently ignoring unsupported characters as it is now. This can cause confusion when troubleshooting a workflow

**Additional context**

**Comment by johanneskoester** [comment]
Yes, a warning is a good idea. Happy to receive a PR! Thanks a lot!
